packages/core/src/render3/instructions/defer.ts

```diff
@@ -332,18 +335,34 @@ function setTDeferBlockDetails(
   tView.data[slotIndex] = deferBlockConfig;
 }
 
+function getTemplateIndexForState(
+    newState: DeferBlockInstanceState, hostLView: LView, tNode: TNode): number|null {
+  const tView = hostLView[TVIEW];
+  const tDetails = getTDeferBlockDetails(tView, tNode);
+
+  if (newState === DeferBlockInstanceState.COMPLETE) {
+    return tDetails.primaryTmplIndex;
+  } else if (newState === DeferBlockInstanceState.LOADING) {
+    return tDetails.loadingTmplIndex;
+  } else if (newState === DeferBlockInstanceState.ERROR) {
+    return tDetails.errorTmplIndex;
+  } else if (newState === DeferBlockInstanceState.PLACEHOLDER) {
+    return tDetails.placeholderTmplIndex;
+  }
+
+  return null;
```

Comment:
I think we can throw an error in this case:
```suggestion

  ngDevMode && throwError(`Unexpected defer block state: ${newState}`);
```